src/app/pages/FastBtcPage/components/Withdraw/AddressForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useCallback, useContext, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import debounce from 'lodash.debounce';
+import { translations } from 'locales/i18n';
+import { WithdrawContext, WithdrawStep } from '../../contexts/withdraw-context';
+import { FormGroup } from 'app/components/Form/FormGroup';
+import { Input } from '../../../../components/Form/Input';
+import { contractReader } from '../../../../../utils/sovryn/contract-reader';
+import { ErrorBadge } from '../../../../components/Form/ErrorBadge';
+import { FastBtcButton } from '../FastBtcButton';
+
+type AddressValidation = 'none' | 'loading' | 'valid' | 'invalid';
+
+export const AddressForm: React.FC = () => {
+  const { address, set } = useContext(WithdrawContext);
+  const { t } = useTranslation();
+
+  const [valid, setValid] = useState<AddressValidation>('none');
+  const [value, setValue] = useState(address);
+
+  const invalid = useMemo(() => valid === 'invalid', [valid]);
+
+  const onContinueClick = useCallback(
+    () =>
+      set(prevState => ({
+        ...prevState,
+        address: value,
+        step: WithdrawStep.REVIEW,
+      })),
+    [set, value],
+  );
+
+  const validate = useCallback(async (adr: string) => {
+    setValid('loading');
+    const result = await contractReader.call(
+      'fastBtcBridge',
+      'isValidBtcAddress',
+      [adr],
+    );
+    setValid(result ? 'valid' : 'invalid');
+  }, []);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const delayedOnChange = useCallback(
+    debounce(adr => validate(adr), 300),
+    [validate],
+  );
+
+  const onChange = useCallback(
```

Comment:
I would call `onChange` (or just `validate`) in `useEffect` on every render because currently, it works the way that if you enter a correct address and go to the next step to review and decide to go back, it will not validate the address until you change the value in the input. That's definitely not user-friendly.